File: ios-samples/CustomSample/CustomSample/KPCustomNotificationView.m
//
//  KPCustomNotificationView.m
//  Kiip
//
//  Created by Ash Bhat on 7/9/13.
//  Copyright (c) 2013 Kiip, Inc. All rights reserved.
//

#import <KiipSDK/KPPoptart.h>

#import "KPCustomNotificationView.h"

#define HEIGHT 46

@implementation KPCustomNotificationView

- (id)initWithFrame:(CGRect)frame {    
    self = [super initWithFrame:frame];
    if (self) {
        self.autoresizingMask = UIViewAutoresizingFlexibleWidth;
        
        self.backgroundColor = [UIColor clearColor];
        self.background = [[UIImage imageNamed:@"button"] stretchableImageWithLeftCapWidth:6 topCapHeight:6];
        
        self.titleFont = [UIFont fontWithName:@"AvenirNext-DemiBold" size:12];
        self.messageFont = [UIFont fontWithName:@"AvenirNext-Medium" size:12];
        
        // Fallbacks
        if (!self.titleFont) {
            self.titleFont = [UIFont fontWithName:@"HelveticaNeue-Bold" size:11];
        }
        if (!self.messageFont) {
            self.messageFont = [UIFont fontWithName:@"HelveticaNeue-Medium" size:11];
        }
    }
    return self;
    
}

- (void) didMoveToSuperview {
    self.frame = CGRectMake(0, 0, CGRectGetWidth(self.superview.bounds), HEIGHT);
    
    self.titleRect = CGRectMake(65, 7, CGRectGetWidth(self.bounds) - 65 - 20 - 10, self.titleFont.lineHeight);
    self.messageRect = CGRectMake(65, 23, CGRectGetWidth(self.bounds) - 65 - 10, self.messageFont.lineHeight);
}

- (void) drawRect:(CGRect)rect {
    KPNotification *notification = self.poptart.notification;
    
    // Background
    [self.background drawInRect:self.bounds];
    
    // Icon
    [notification.icon drawInRect:CGRectMake(5, 2, 55, 42)];
    
    // Title
    [[UIColor whiteColor] set];
    [notification.title drawInRect:self.titleRect withFont:self.titleFont lineBreakMode:UILineBreakModeTailTruncation];
    
    // Message
    [[UIColor whiteColor] set];
    [notification.message drawInRect:self.messageRect withFont:self.messageFont lineBreakMode:UILineBreakModeTailTruncation];
}

@end

File: ios-samples/CustomSwiftSample/CustomSwiftSample/KPCustomNotificationView.swift
//
//  KPCustomNotificationView.swift
//  CustomSwiftSample
//
//  Created by Sahil Verma on 7/28/14.
//  Copyright (c) 2014 kiip. All rights reserved.
//

import Foundation

class KPCustomNotificationView : KPNotificationView {

    // properties describing the custom notification view.
    var background : UIImage
    var titleFont : UIFont
    var messageFont : UIFont
    var titleRect : CGRect
    var messageRect : CGRect

    let HEIGHT : CGFloat = 46

    init() {
        titleFont = UIFont(name:"HelveticaNeue-Bold", size: 12)
        messageFont = UIFont(name:"HelveticaNeue-Medium", size: 12)
        var backgroundColor = UIColor.clearColor()
        background = UIImage(named:"button.png")
        titleRect = CGRect()
        messageRect = CGRect()
        super.init(frame: CGRectZero)
        self.autoresizingMask = UIViewAutoresizing.FlexibleWidth
    }

    override func didMoveToSuperview() {
        if self.superview == nil {
            return
        }
        self.frame = CGRectMake(0, 0, CGRectGetWidth(self.superview.bounds), HEIGHT)
        titleRect = CGRectMake(65, 7, CGRectGetWidth(self.bounds) - 65 - 20 - 10, titleFont.lineHeight)
        messageRect = CGRectMake(65, 24, CGRectGetWidth(self.bounds) - 65 - 10, messageFont.lineHeight)
    }

    override func drawRect(rect: CGRect) -> Void {

        var notification:KPNotification = self.poptart.notification

        // background
        background.drawInRect(self.bounds)

        // Icon
        notification.icon.drawInRect(CGRectMake(5, 2, 55, 42))

        // Title
        var attrs = NSMutableDictionary()
        attrs.setValue(titleFont, forKey:NSFontAttributeName)
        // TODO
        //attrs.setValue(NSLineBreakMode.ByTruncatingTail, forKey:lineBreakMode)

        UIColor.whiteColor().set()
        notification.title.drawInRect(titleRect, withAttributes: attrs)
        
        // Message
        notification.message.drawInRect(messageRect, withAttributes: attrs)
    }
}